---
title: "Methods"
author: "Evgenii O. Tretiakov"
date: "`r Sys.Date()`"
output:
  word_document:
  workflowr::wflow_html:
    toc: true
    toc_float: yes
    theme: cosmo
    highlight: textmate
    code_folding: hide
    df_print: paged
editor_options: 
  markdown: 
    wrap: 72
bibliography:
    - "`r here::here('data/references/references.bib')`"
    - "`r here::here('output/methods/packages.bib')`"
---

```{r knitr, include = FALSE}
DOCNAME = "methods"
NOW <- Sys.time()

# Time chunks during knitting
knitr::knit_hooks$set(timeit = function(before) {
    if (before) {
        print(paste("Start:", Sys.time()))
        NOW <<- Sys.time()
    } else {
        print(paste("Stop:", Sys.time()))
        print(Sys.time() - NOW)
    }
})

knitr::opts_chunk$set(
    autodep        = TRUE,
    cache          = FALSE,
    cache.path     = paste0("cache/", DOCNAME, "/"),
    cache.comments = FALSE,
    echo           = TRUE,
    error          = FALSE,
    fig.align      = "center",
    fig.width      = 10,
    fig.height     = 8,
    message        = FALSE,
    warning        = FALSE,
    timeit         = TRUE
)
```

```{r libaries, cache = FALSE}
# Presentation
library("glue")
library("knitr")

# JSON
library("jsonlite")

# Tidyverse
library("tidyverse")
```

```{r source, cache = FALSE}

```

```{r pkg-bib}
dir.create(here::here("output", DOCNAME), showWarnings = FALSE)

write_bib(c("base", "Seurat", "SeuratWrappers", "SeuratDisk", "sctransform",
            "glmGamPoi", "patchwork", "scCustomize", "Nebulosa", "clustree",
            "mrtree", "gprofiler2", "cowplot", "UpSetR", "ggstatsplot",
            "gridExtra", "tidyverse", "dplyr", "tidyr", "magrittr", "stringr",
            "skimr", "future", "purrr", "here", "workflowr", "zeallot", "knitr",
            "kableExtra", "rmarkdown", "reticulate"),
          file = here::here("output", DOCNAME, "packages.bib"))
```

```{r load}
versions <- list(
    biomaRt        = packageVersion("biomaRt"),
    cellbender     = "docker://etretiakov/cellbender:v0.0.1",
    cellranger     = "7.1.0",
    cellranger_ref = "mm10_optimized_v.1.0",
    clustree       = packageVersion("clustree"),
    cowplot        = packageVersion("cowplot"),
    dplyr          = packageVersion("dplyr"),
    future         = packageVersion("future"),
    ggplot2        = packageVersion("ggplot2"),
    ggstatsplot    = packageVersion("ggstatsplot"),
    glmGamPoi      = packageVersion("glmGamPoi"),
    gprofiler2     = packageVersion("gprofiler2"),
    gridExtra      = packageVersion("gridExtra"),
    here           = packageVersion("here"),
    kableExtra     = packageVersion("kableExtra"),
    knitr          = packageVersion("knitr"),
    magrittr       = packageVersion("magrittr"),
    mrtree         = packageVersion("mrtree"),
    Nebulosa       = packageVersion("Nebulosa"),
    pandoc         = rmarkdown::pandoc_version(),
    patchwork      = packageVersion("patchwork"),
    purrr          = packageVersion("purrr"),
    python         = "3.8.8",
    R              = str_extract(R.version.string, "[0-9\\.]+"),
    reticulate     = packageVersion("reticulate"),
    rmarkdown      = packageVersion("rmarkdown"),
    scCustomize    = packageVersion("scCustomize"),
    sctransform    = packageVersion("sctransform"),
    Seurat         = packageVersion("Seurat"),
    skimr          = packageVersion("skimr"),
    stringr        = packageVersion("stringr"),
    Snakemake      = "7.21.0",
    souporcell     = "shub://wheaton5/souporcell",
    tidyr          = packageVersion("tidyr"),
    tidyverse      = packageVersion("tidyverse"),
    UpSetR         = packageVersion("UpSetR"),
    viridis        = packageVersion("viridis"),
    workflowr      = packageVersion("workflowr"),
    zeallot        = packageVersion("zeallot")
)
```

# Pre-processing

The Cell Ranger pipeline (v`r versions$cellranger`)
[@zhengMassivelyParallelDigital2017] was used to perform sample
demultiplexing, barcode processing and single-nuclei gene counting.
Briefly, samples were demultiplexed to produce a pair of FASTQ files for
each sample. Reads containing sequence information were aligned using
the optimised mouse genome reference (v`r versions$cellranger_ref`)
provided by Pool's lab based on the default Cell Ranger mm10 genome
version 2020-A that was cleared from gene overlaps, poorly annotated
exons and 3'-UTRs and intergenic fragments
[@poolEnhancedRecoverySinglecell2022]. PCR duplicates were removed by
selecting unique combinations of cell barcodes, unique molecular
identifiers (UMIs) and gene ids with the final results being a gene
expression matrix that was used for further analysis.

# Quality control

## Droplet selection

The droplet selection method of Cell Ranger identified 923 nuclei in
ventrobasal thalamus and in principal sensory trigeminal nucleus - 731
nuclei based on EmptyDrops method
[@lunEmptyDropsDistinguishingCells2019] incorporated into cellranger
count pipeline.

## Ambient RNA removal

Using those values as the expected number of cells, we applied a neural
network-based approach called CellBender (`r versions$cellbender`)
[@fleming2022]. We set a false positive rate threshold at the level of
0.01 and set the neural network to learn over 150 epochs with a total of
5000 droplets included based on knee plots (please see online
supplementary Cell Ranger reports).

## Doublets detection

We quantified log probability to be a doublet for every cell based on
apriori knowledge of genotypes from each input sample and called variant
occurrence frequencies that allowed to cluster nuclei to source organism
or classify as a doublet [@heatonSouporcellRobustClustering2020] (see
`r versions$souporcell`).

## Further filtering

Gene annotation information was added using the gprofiler2 package
(v`r versions$gprofiler2`) [@reimandProfileraWebServer2016]; thus we
filter cells based on high content of mitochondrial, ribosomal or
hemoglobin proteins genes, specifically thresholds set as 1%, 1% and
0.5%; additionally pseudogenes and poorly annotated genes were also
deleted from count matrix. Moreover, cells of low complexity were
filtered out as ($\log_{10}Genes/\log_{10}UMI < 0.8$). Therefore, cells
were assigned cell cycle scores using the `CellCycleScoring` function in
the Seurat package (v`r versions$Seurat`).

# Clustering

## Gene selection

We used the selection method in the Seurat package
(v`r versions$Seurat`) [@satijaSpatialReconstructionSinglecell2015;
@stuartIntegrativeSinglecellAnalysis2019], which uses a modern variance
stabilising transformation statistical technic that utilises scaling to
person residuals [@hafemeister2019]. That way, we selected 3000 highly
variable genes per dataset and regressed out complexity and cell-cycle
variability prior to the final scaling of filtered matrixes.

## Graph-based and multi-level reconcile tree clustering

We performed Leiden algorithm graph-based clustering. PCA was performed
using the selected genes and the jackknife tested
[@chungStatisticalSignificanceVariables2015] principal components (we
tested the significance of feature for randomly picked 1% of data over
1000 iterations; see `PCScore` function in `functions.R` script of code
directory) were used to construct a shared nearest neighbour graph using
the overlap between the `15` nearest neighbours of each cell. Leiden
modularity optimisation [@traag2019] was used to partition this graph
with an array of resolution parameters where 30 modularity events were
sampled between `0.2` and `2.5`. Clustering tree visualisations
[@zappiaClusteringTreesVisualization2018] were produced using the
clustree package (v`r versions$clustree`) showing the resolution of
previously identified clusters. By inspecting these resolutions
reconcile tree produced by mrtree package (v`r versions$mrtree`)
[@pengCellTypeHierarchy2021] and calculating adjusted multi-resolution
Rand index chosen as maximum value if there is no higher modularity
within `0.05` AMRI difference (see `SelectResolution` in `function.R`
file of code directory).

# Marker genes

Marker genes for each cluster were identified using logreg test
[@ntranos2019] implemented in Seurat framework (v`r versions$seurat`)
[@stuartIntegrativeSinglecellAnalysis2019]. Genes were considered
significant markers for a cluster if they had an FDR less than `0.001`.
Identities were assigned to each cluster by comparing the detected genes
to previously published markers and our own validation experiments.

# Other packages

Visualisations and figures were primarily created using the ggplot2
(v`r versions$ggplot2`), cowplot (v`r versions$cowplot`) [@R-cowplot]
and patchwork (v`r versions$patchwork`) packages using the viridis
colour palette (v`r versions$viridis`) for continuous data. UpSet plots
[@conwayUpSetRPackageVisualization2017] were produced using the UpSetR
package (v`r versions$UpSetR`) [@R-UpSetR] with help from the gridExtra
package (v`r versions$gridExtra`) [@R-gridExtra]. Data manipulation was
performed using other packages in the tidyverse
(v`r versions$tidyverse`) [@R-tidyverse] particularly dplyr
(v`r versions$dplyr`) [@R-dplyr], tidyr (v`r versions$dplyr`) [@R-tidyr]
and purrr (v`r versions$purrr`) [@R-purrr]. The analysis project was
managed using the Snakemake system (v `r versions$snakemake`)
[@molderSustainableDataAnalysis2021] and the workflowr
(v`r versions$workflowr`) [@R-workflowr] package which was also used to
produce the publicly available website displaying the analysis code,
results and output. Reproducible reports were produced using knitr
(v`r versions$knitr`) [@R-knitr] and R Markdown
(v`r versions$rmarkdown`) [@R-rmarkdown] and converted to HTML using
Pandoc (v`r versions$pandoc`).

# Summary

## Output files

```{r save}
versions <- purrr::map(versions, as.character)
versions <- jsonlite::toJSON(versions, pretty = TRUE)
readr::write_lines(versions,
                   here::here("output", DOCNAME, "package-versions.json"))
```

# References

